//
// Copyright 2017 Asylo authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

package asylo.test;

// This proto file defines three services--Messenger1, Messenger2, and
// Messenger3--that are replicas of each other. Each of these services is very
// simple, and defines exactly one RPC. The three replicas are defined to enable
// testing of gRPC servers hosting multiple services.

// Messenger1 is a simple service for testing various Asylo gRPC features.
service Messenger1 {
  // Hello is a simple RPC that requires the user submit a |name| in order to
  // get a |message|.
  rpc Hello(HelloRequest) returns (HelloResponse) {
  }
}

// Messenger2 is a simple service for testing various Asylo gRPC features.
service Messenger2 {
  // Hello is a simple RPC that requires the user submit a |name| in order to
  // get a |message|.
  rpc Hello(HelloRequest) returns (HelloResponse) {
  }
}

// Messenger3 is a simple service for testing various Asylo gRPC features.
service Messenger3 {
  // Hello is a simple RPC that requires the user submit a |name| in order to
  // get a |message|.
  rpc Hello(HelloRequest) returns (HelloResponse) {
  }
}

message HelloRequest {
  optional string name = 1;
}

message HelloResponse {
  optional string message = 1;
}
